### PR TITLE
Add system tests for site and global data

### LINKED ISSUE #125: Add system tests for site and global data
Methods to test include:
set_global_data
get_global_data
global_data_exists
set_site_data
get_site_data
site_data_exists

### PR DESCRIPTION
**Pull request by smooresni**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nitsm-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Adds system tests for setting and querying site and global data

### Why should this Pull Request be merged?

Resolves #125 

### What testing has been done?

Full suite.

- [x] I have run the automated tests (required if there are code changes)

### CODE REVIEW

File: systemtests/site_and_global_data_codemodules.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    tsm_context.set_site_data(data_id, data)
+
+
+@nitsm.codemoduleapi.code_module
+def check_site_data(tsm_context: SemiconductorModuleContext, data_id, data):
+    site_count = len(tsm_context.site_numbers)
+    site_data_exists = [tsm_context.site_data_exists(data_id)] * site_count
+    tsm_context.publish_per_site(site_data_exists, "SiteDataExists")
+
+    valid_site_data = [tsm_context.get_site_data(data_id) == tuple(data)] * site_count
```
Comment (ni-erikcrank) [inline systemtests/site_and_global_data_codemodules.py:31]: Why is it necessary to use `tuple` here? Isn't `data` a tuple?

Comment (dxie-dxie) [reply systemtests/site_and_global_data_codemodules.py:31]: `data `is of class 'list'

**Review by ni-erikcrank — APPROVED** [review]

**Pull request merged**